### PR TITLE
Initial playground

### PR DESCRIPTION
**Pull request by thorstendb-ARM**
Start of project
https://github.com/Open-CMSIS-Pack/generator-bridge/issues/7

### CODE REVIEW

**Comment by thorstendb-ARM** [comment]
Ready for review 0.9
Remarks:
- Linter: Error: Error return value is not checked (errcheck)
  - need hint
- utils_test: TestFileExists()
  - Test runs when "Debug" but fails in "Run"
-

**Comment by soumeh01** [comment]
> Ready for review 0.9 Remarks:
> 
> * Linter: Error: Error return value is not checked (errcheck)
>   
>   * need hint
> * utils_test: TestFileExists()
>   
>   * Test runs when "Debug" but fails in "Run"
> * Vulnerability No1: GO-2023-2185:
>   
>   * Error:       No1: internal/stm32CubeMX/stm32CubeMX.go:107:16: stm32CubeMX.Launch calls exec.Cmd.Run, which eventually calls filepath.Abs
>   * Error:       No2: internal/stm32CubeMX/stm32CubeMX.go:101:21: stm32CubeMX.Launch calls exec.Command, which calls filepath.Base
>   * Error:       No3: internal/utils/utils.go:70:23: utils.ConvertFilename calls filepath.Clean
>   * Error:       No4: internal/readFile/readFile.go:22:25: readFile.Process calls filepath.Dir
>   * Error:       No5: internal/stm32CubeMX/stm32CubeMX.go:116:27: stm32CubeMX.WriteProjectFile calls filepath.Join
>   * Error:       No6: internal/utils/utils.go:82:26: utils.ConvertFilename calls filepath.Rel
>   * Error:       No7: internal/stm32CubeMX/stm32CubeMX.go:107:16: stm32CubeMX.Launch calls exec.Cmd.Run, which eventually calls filepath.VolumeName

Linter error: Please check the[ summary page](https://github.com/Open-CMSIS-Pack/generator-bridge/actions/runs/6847073854). Github reports 11 issues with code annotation.
Vulnerability: AS the error states, Please update the golang version to 1.21 as the older version 1.20 has vulnerabilities

File: cmd/commands/commands.go [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+	inFile  string
+	inFile2 string
+	outPath string
+}
+
+var Version string
+var CopyRight string
+
+func printVersionAndLicense(file io.Writer) {
+	fmt.Fprintf(file, "generator-bridge version %v %s\n", strings.ReplaceAll(Version, "v", ""), CopyRight)
```
Comment (jkrech) [inline cmd/commands/commands.go:65]: Copyright is a term/noun and not constructed from copy and right. Therefore camelcasing does not seem right.

File: cmd/commands/commands.go [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+	}
+
+	return nil
+}
+
+var flags struct {
+	version bool
+	help    bool
+	inFile  string
+	inFile2 string
```
Comment (jkrech) [inline cmd/commands/commands.go:57]: What is the inFile2 for?

**Review by jkrech — COMMENTED** [review]
I am unable to give qualified comments. But the second input file is not clear to me.

File: cmd/commands/commands.go (hunk 2 above)
Comment (thorstendb-ARM) [reply cmd/commands/commands.go:57]: used for command "--read" which is used for generating files from pre-generated CubeMX:
```
            //"args": ["--read", "D:/cubemx/ref2/1/2/3/4/noTZ/test.cbuild-gen-idx.yml", "-f", "D:/cubemx/ref2/1/STM32CubeMX/STM32U585AIIx/STM32CubeMX/.mxproject"]
            //"args": ["--read", "D:/cubemx/ref2/1/STM32CubeMX/STM32U585AIIx/STM32CubeMX/.mxproject", "--out", "D:/cubemx/ref2/1/STM32CubeMX/STM32U585AIIx"]

```

Comment (jkrech) [reply cmd/commands/commands.go:57]: It is my expectation that the `*.cbuild-gen-idx.yml` is the only parameter being passed to the bridge.
Based on the generator directory provided, the tool needs to check whether an ioc file is present or not and where the outputfile needs to reside, what the working directory is, etc.
What am I missing?

**Review by soumeh01 — APPROVED** [review]
lgtm

**Review by brondani — APPROVED** [review]
Tested with [cgen_mockup](https://github.com/brondani/cgen_mockup/tree/csolution_poc) examples.
LGTM

**Review by jkrech — APPROVED** [review]
LGTM as far as I understand it.

**Pull request merged**